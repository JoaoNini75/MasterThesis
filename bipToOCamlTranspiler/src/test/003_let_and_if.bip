let max (arg1 : int, arg2 : int) : int = begin
	if arg1 > arg2
	then begin arg1 end
	else begin arg2 end
end
(*@ ensures true *)

let fun2 (arg1, arg2) = begin
	let a = 2 in
	let b = ref (a * 10) in

	let c = 
		if arg1 + a == arg2
		then begin arg1 * a - !b end
		else begin arg2 end
	in

	a mod c * !b
end
(*@ ensures true *)

let unitif (x) = begin
	let a = ref (-2) in
	if !a > x
	then begin () end
	else begin () end
end
(*@ ensures true *)

let p2 (arg) = begin
  let x = ref 1 in
  let y = ref arg in
	x := 0;
  x := !x * !y;

  if !y > 0
	then begin 
		y := !x * 10;
		x := !x + arg
	end else begin
		x := !y * arg;
		y := !x + 10
	end;

  !x + !y
end
(*@ ensures true *)
