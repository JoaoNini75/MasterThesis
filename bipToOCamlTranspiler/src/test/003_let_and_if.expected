let max (arg1 : int) (arg2 : int) : int =
  if arg1 > arg2
  then begin 
    arg1
  end else begin 
    arg2
  end
(*@ ensures true *)

let fun2 (arg1) (arg2) =
  let a = 2 in
  let b = ref (a * 10) in
  let c = 
    if arg1 + a = arg2
    then begin 
      arg1 * a - !b
    end else begin 
      arg2
    end in
  a mod c * !b
(*@ ensures true *)

let unitif (x) =
  let a = ref (-2) in
  if !a > x
  then begin 
    ()
  end else begin 
    ()
  end
(*@ ensures true *)

let p2 (arg) =
  let x = ref (1) in
  let y = ref (arg) in
  x := 0;
  x := !x * !y;
  if !y > 0
  then begin 
    y := !x * 10;
    x := !x + arg
  end else begin 
    x := !y * arg;
    y := !x + 10
  end;
  !x + !y
(*@ ensures true *)


